xml/object.go

```diff
@@ -0,0 +1,73 @@
+package xml
+
+import (
+	"bytes"
+)
+
+// Object represents the encoding of a XML Object type
+type Object struct {
+	w       *bytes.Buffer
+	scratch *[]byte
+}
+
+// newObject returns a new object encoder type
+func newObject(w *bytes.Buffer, scratch *[]byte) *Object {
+	return &Object{w: w, scratch: scratch}
+}
+
+// Key returns a Value encoder that should be used to encode a XML value type.
+// It sets the given named key builder function into the XML object value encoder.
+// Key takes optional functional arguments to set tag metadata when building the element tag.
+func (o *Object) Key(name string, opts ...func(t *TagMetadata)) Value {
+	// openTagFn is a element start tag builder function
+	var openTagFn = func() {
+		writeOpenTag(o.w, name)
+		defer o.w.WriteRune(rightAngleBracket)
+		for _, fn := range opts {
+			var m TagMetadata
+			fn(&m)
+
+			// set the namespace URI and prefix in element tag
+			if len(m.NamespacePrefix) != 0 && len(m.NamespaceURI) != 0 {
+				o.w.Write([]byte(" xmlns:" + m.NamespacePrefix + "=\"" + m.NamespaceURI + "\""))
+			}
```

Comment:
This looks like it forces the user of the Encoder to keep track of name spacing for nested attributes and tags. Is that intended?

Reply:
Also attributes and namespaces could be applied to any XML node such as our Maps and Lists, I'm not sure how that works with Smithy's modelling, but is this something that should be supported for all node groupings?